gen/media/audio/pulse/pulse_stubs.cc:155:27: error: conflicting types for 'pa_context_get_state'
extern pa_context_state_t pa_context_get_state(pa_context* c) __attribute__((weak));
                          ^
../../../../../../../../../../usr/local/include/pulse/context.h:199:20: note: previous declaration is here
pa_context_state_t pa_context_get_state(const pa_context *c);
                   ^
gen/media/audio/pulse/pulse_stubs.cc:156:21: error: conflicting types for 'pa_context_get_state'
pa_context_state_t  pa_context_get_state(pa_context* c) {
                    ^
../../../../../../../../../../usr/local/include/pulse/context.h:199:20: note: previous declaration is here
pa_context_state_t pa_context_get_state(const pa_context *c);
                   ^
gen/media/audio/pulse/pulse_stubs.cc:171:29: error: conflicting types for 'pa_operation_get_state'
extern pa_operation_state_t pa_operation_get_state(pa_operation* o) __attribute__((weak));
                            ^
../../../../../../../../../../usr/local/include/pulse/operation.h:52:22: note: previous declaration is here
pa_operation_state_t pa_operation_get_state(const pa_operation *o);
                     ^
gen/media/audio/pulse/pulse_stubs.cc:172:23: error: conflicting types for 'pa_operation_get_state'
pa_operation_state_t  pa_operation_get_state(pa_operation* o) {
                      ^
../../../../../../../../../../usr/local/include/pulse/operation.h:52:22: note: previous declaration is here
pa_operation_state_t pa_operation_get_state(const pa_operation *o);
                     ^
gen/media/audio/pulse/pulse_stubs.cc:211:17: error: conflicting types for 'pa_stream_get_device_index'
extern uint32_t pa_stream_get_device_index(pa_stream* s) __attribute__((weak));
                ^
../../../../../../../../../../usr/local/include/pulse/stream.h:366:10: note: previous declaration is here
uint32_t pa_stream_get_device_index(const pa_stream *s);
         ^
gen/media/audio/pulse/pulse_stubs.cc:212:11: error: conflicting types for 'pa_stream_get_device_index'
uint32_t  pa_stream_get_device_index(pa_stream* s) {
          ^
../../../../../../../../../../usr/local/include/pulse/stream.h:366:10: note: previous declaration is here
uint32_t pa_stream_get_device_index(const pa_stream *s);
         ^
gen/media/audio/pulse/pulse_stubs.cc:219:26: error: conflicting types for 'pa_stream_get_state'
extern pa_stream_state_t pa_stream_get_state(pa_stream* p) __attribute__((weak));
                         ^
../../../../../../../../../../usr/local/include/pulse/stream.h:345:19: note: previous declaration is here
pa_stream_state_t pa_stream_get_state(const pa_stream *p);
                  ^
gen/media/audio/pulse/pulse_stubs.cc:220:20: error: conflicting types for 'pa_stream_get_state'
pa_stream_state_t  pa_stream_get_state(pa_stream* p) {
                   ^
../../../../../../../../../../usr/local/include/pulse/stream.h:345:19: note: previous declaration is here
pa_stream_state_t pa_stream_get_state(const pa_stream *p);
                  ^
gen/media/audio/pulse/pulse_stubs.cc:235:12: error: conflicting types for 'pa_proplist_contains'
extern int pa_proplist_contains(pa_proplist* p, const char* key) __attribute__((weak));
           ^
../../../../../../../../../../usr/local/include/pulse/proplist.h:390:5: note: previous declaration is here
int pa_proplist_contains(const pa_proplist *p, const char *key);
    ^
gen/media/audio/pulse/pulse_stubs.cc:236:6: error: conflicting types for 'pa_proplist_contains'
int  pa_proplist_contains(pa_proplist* p, const char* key) {
     ^
../../../../../../../../../../usr/local/include/pulse/proplist.h:390:5: note: previous declaration is here
int pa_proplist_contains(const pa_proplist *p, const char *key);
    ^
gen/media/audio/pulse/pulse_stubs.cc:243:20: error: conflicting types for 'pa_proplist_gets'
extern const char* pa_proplist_gets(pa_proplist* p, const char* key) __attribute__((weak));
                   ^
../../../../../../../../../../usr/local/include/pulse/proplist.h:314:13: note: previous declaration is here
const char *pa_proplist_gets(const pa_proplist *p, const char *key);
            ^
gen/media/audio/pulse/pulse_stubs.cc:244:14: error: conflicting types for 'pa_proplist_gets'
const char*  pa_proplist_gets(pa_proplist* p, const char* key) {
             ^
../../../../../../../../../../usr/local/include/pulse/proplist.h:314:13: note: previous declaration is here
const char *pa_proplist_gets(const pa_proplist *p, const char *key);
            ^
gen/media/audio/pulse/pulse_stubs.cc:251:15: error: conflicting types for 'pa_stream_readable_size'
extern size_t pa_stream_readable_size(pa_stream *p) __attribute__((weak));
              ^
../../../../../../../../../../usr/local/include/pulse/stream.h:574:8: note: previous declaration is here
size_t pa_stream_readable_size(const pa_stream *p);
       ^
gen/media/audio/pulse/pulse_stubs.cc:252:9: error: conflicting types for 'pa_stream_readable_size'
size_t  pa_stream_readable_size(pa_stream *p) {
        ^
../../../../../../../../../../usr/local/include/pulse/stream.h:574:8: note: previous declaration is here
size_t pa_stream_readable_size(const pa_stream *p);
       ^
gen/media/audio/pulse/pulse_stubs.cc:279:12: error: conflicting types for 'pa_context_errno'
extern int pa_context_errno(pa_context *c) __attribute__((weak));
           ^
../../../../../../../../../../usr/local/include/pulse/context.h:193:5: note: previous declaration is here
int pa_context_errno(const pa_context *c);
    ^
gen/media/audio/pulse/pulse_stubs.cc:280:6: error: conflicting types for 'pa_context_errno'
int  pa_context_errno(pa_context *c) {
     ^
../../../../../../../../../../usr/local/include/pulse/context.h:193:5: note: previous declaration is here
int pa_context_errno(const pa_context *c);
    ^

--- media/audio/pulse/pulse.sigs.orig	2019-03-19 08:52:48 UTC
+++ media/audio/pulse/pulse.sigs
@@ -24,11 +24,11 @@ pa_operation* pa_context_get_source_info_by_index(pa_c
 pa_operation* pa_context_get_source_info_by_name(pa_context* c, const char* name, pa_source_info_cb_t cb, void *userdata);
 pa_operation* pa_context_get_source_info_list(pa_context* c, pa_source_info_cb_t cb, void* userdata);
 pa_operation* pa_context_get_sink_info_list(pa_context* c, pa_sink_info_cb_t cb, void* userdata);
-pa_context_state_t pa_context_get_state(pa_context* c);
+pa_context_state_t pa_context_get_state(const pa_context *c);
 pa_context* pa_context_new(pa_mainloop_api* mainloop, const char* name);
 pa_operation* pa_context_set_source_volume_by_index(pa_context* c, uint32_t idx, const pa_cvolume* volume, pa_context_success_cb_t cb, void* userdata);
 void pa_context_set_state_callback(pa_context* c, pa_context_notify_cb_t cb, void* userdata);
-pa_operation_state_t pa_operation_get_state(pa_operation* o);
+pa_operation_state_t pa_operation_get_state(const pa_operation* o);
 void pa_context_unref(pa_context* c);
 void pa_operation_unref(pa_operation* o);
 int pa_stream_begin_write(pa_stream* p, void** data, size_t* nbytes);
@@ -38,23 +38,23 @@ pa_operation* pa_stream_cork(pa_stream* s, int b, pa_s
 int pa_stream_disconnect(pa_stream* s);
 int pa_stream_drop(pa_stream *p);
 pa_operation* pa_stream_flush(pa_stream* s, pa_stream_success_cb_t cb, void* userdata);
-uint32_t pa_stream_get_device_index(pa_stream* s);
+uint32_t pa_stream_get_device_index(const pa_stream* s);
 int pa_stream_get_latency(pa_stream* s, pa_usec_t* r_usec, int* negative);
-pa_stream_state_t pa_stream_get_state(pa_stream* p);
+pa_stream_state_t pa_stream_get_state(const pa_stream* p);
 pa_stream* pa_stream_new(pa_context*  c, const char*  name, const pa_sample_spec*  ss, const pa_channel_map *  map);
 pa_stream* pa_stream_new_with_proplist(pa_context* c, const char* name, const pa_sample_spec* ss, const pa_channel_map* map, pa_proplist* p);
 pa_proplist* pa_proplist_new(void);
-int pa_proplist_contains(pa_proplist* p, const char* key);
+int pa_proplist_contains(const pa_proplist* p, const char* key);
 void pa_proplist_free(pa_proplist* p);
-const char* pa_proplist_gets(pa_proplist* p, const char* key);
+const char* pa_proplist_gets(const pa_proplist* p, const char* key);
 int pa_proplist_sets(pa_proplist* p, const char* key, const char* value);
-size_t pa_stream_readable_size(pa_stream *p);
+size_t pa_stream_readable_size(const pa_stream *p);
 int pa_stream_peek(pa_stream* p, const void** data, size_t* nbytes);
 void pa_stream_set_read_callback(pa_stream* p, pa_stream_request_cb_t cb, void* userdata);
 void pa_stream_set_state_callback(pa_stream* s, pa_stream_notify_cb_t cb, void* userdata);
 int pa_stream_write(pa_stream* p, const void* data, size_t nbytes, pa_free_cb_t free_cb, int64_t offset, pa_seek_mode_t seek);
 void pa_stream_set_write_callback(pa_stream *p, pa_stream_request_cb_t cb, void *userdata);
 void pa_stream_unref(pa_stream* s);
-int pa_context_errno(pa_context *c);
+int pa_context_errno(const pa_context *c);
 const char* pa_strerror(int error);
 pa_cvolume* pa_cvolume_set(pa_cvolume* a, unsigned  channels, pa_volume_t v);
